### PR TITLE
Update retrying client documentation

### LINKED ISSUE #309: Update Retrying Tarantool client documentation

### PR DESCRIPTION
**Pull request by iDneprov**
Update retrying client documentation

I haven't forgotten about:
- [x] Tests
- [ ] Changelog
- [x] Documentation
- [x] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:

Closes #309

### CODE REVIEW

File: docs/RetryingTarantoolClient.md [hunk 1]
```diff
@@ -11,38 +11,14 @@ By default, failed requests will be repeated only for some known network problem
...
+
+You can set up basic client
+https://github.com/tarantool/cartridge-java/blob/ba5602651a89132ead5f33f00a1a4462bce117d0/src/test/java/io/tarantool/driver/integration/ReconnectIT.java#L109-L123
+And reuse it then you need retrying client
+https://github.com/tarantool/cartridge-java/blob/ba5602651a89132ead5f33f00a1a4462bce117d0/src/test/java/io/tarantool/driver/integration/ReconnectIT.java#L153-L171
+But you don't have to set up basic client if you need retying client only.
+All methods of client builder with prefix `withRetrying` can be used with `createClient`.
+  
+Example of client API usage if you specify using the default CRUD proxy operations mapping configuration.
+https://github.com/tarantool/cartridge-java/blob/8a880423da1ce2bc0e82557d70ab46c9e7eba618/src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java#L64-L107
```
Comment (ArtDu) [inline docs/RetryingTarantoolClient.md:24]: You can have a more visual example and complex example for it cause it's more complex API:
```
...
.withProxyMethodMapping(builder -> builder.withDeleteFunctionName("delete_with_error_if_not_found"))
.withRetryingByNumberOfAttempts(5, 
             // You can use default predicates from TarantoolRequestRetryPolicies for checking errors 
             retryNetworkErrors() 
                 // Also you can use your own predicates and combine them with each other 
                 .or(e -> e.getMessage().contains("Unsuccessful attempt")) 
                 // Or with defaults 
                 .or(retryTarantoolNoSuchProcedureErrors())
                 .or(retryDeleteError()), 
             // Also you can set delay in millisecond between attempts 
             factory -> factory.withDelay(300) 
         ) 
...


space(...).delete(); // retrying 
space(...).insert();
```

File: src/test/java/io/tarantool/driver/integration/ReconnectIT.java [hunk 2]
```diff
@@ -66,21 +77,54 @@ public static void setUp() throws TimeoutException {
...
+
+        // Start another thread that will insert value after 100 ms
+        new Thread(() -> {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            space.insert(tupleFactory.create(1, null, "FIO", 50, 100)).join();
+        }).start();
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/integration/ReconnectIT.java:103]: You dont' have to use Thread API, you can't guarantee that your insert method will be called after deletion. It'd better to use CompletableFuture API, for e.g.
```java
AtomicInteger retryingCounter = new...
...
.or(e -> {
    retryingCounter.addAndGet();
    return e.getMessage().contains("Records not found"); 
})
...
CompletableFuture deleteFuture = retryingClient.space(SPACE_NAME).delete(conditions);
retry(() -> assertEquals(1, retryingCounter.get()))
space.insert(tupleFactory.create(1, null, "FIO", 50, 100)).get();
deleteFuture.get();
```

But for this you need it in TarantoolUtils.java
```java
public static Integer DEFAULT_RETRYING_ATTEMPTS = 5;
    public static Integer DEFAULT_RETRYING_DELAY = 100;

    public static void retry(Runnable fn) throws InterruptedException {
        retry(DEFAULT_RETRYING_ATTEMPTS, DEFAULT_RETRYING_DELAY, fn);
    }

    public static void retry(Integer attempts, Integer delay, Runnable fn) throws InterruptedException {
        while (attempts > 0) {
            try {
                fn.run();
                return;
            } catch (AssertionError ignored) {
            }

            --attempts;
            Thread.sleep(delay);
        }
        fn.run();
    }
```

**Review by ArtDu — APPROVED** [review]
I still don't believe that it works from the first attempt, but good for you :)

File: docs/RetryingTarantoolClient.md [hunk 3]
```diff
@@ -11,38 +11,17 @@ By default, failed requests will be repeated only for some known network problem
...
-
-    TarantoolClient<TarantoolTuple, TarantoolResult<TarantoolTuple>> client = setupClient();
-    String result = retrying(client, 4, 500).callForSingleResult("retrying_function", String.class).get();
-    assertEquals("Success", result);
-    result = retrying(client, 2, 1000).callForSingleResult("retrying_function", String.class).get();
-    assertEquals("Success", result);
-```
+connection pool and basic client settings, and only augment the behavior of the client.  
+
+In this example I use custom delete function.
```
Comment (akudiyar) [inline docs/RetryingTarantoolClient.md:16]: "I use" -> "we use" looks better I believe

Comment (akudiyar) [reply docs/RetryingTarantoolClient.md:16]: Or "let's consider a custom delete function"

**Review by akudiyar — CHANGES_REQUESTED** [review]

File: docs/RetryingTarantoolClient.md (hunk 3 above)
Comment (iDneprov) [reply docs/RetryingTarantoolClient.md:16]: > "I use" -> "we use" looks better I believe

I don't think so

Comment (iDneprov) [reply docs/RetryingTarantoolClient.md:16]: > Or "let's consider a custom delete function"

This looks better. I'll increase the formality of language in documentation soon #356

**Pull request merged**